File: lib/js/__tests__/Json_test.js
'use strict';

var Jest = require("bs-jest/lib/js/src/jest.js");
var Json = require("../src/Json.js");

describe("parse", (function () {
        Jest.test("success", (function () {
                return Jest.Expect[/* toEqual */12](/* Some */[null])(Jest.Expect[/* expect */0](Json.parse("null")));
              }));
        return Jest.test("error", (function () {
                      return Jest.Expect[/* toEqual */12](/* None */0)(Jest.Expect[/* expect */0](Json.parse("{")));
                    }));
      }));

describe("parseOrRaise", (function () {
        Jest.test("success", (function () {
                return Jest.Expect[/* toEqual */12](null)(Jest.Expect[/* expect */0](Json.parseOrRaise("null")));
              }));
        return Jest.test("error", (function () {
                      return Jest.Expect[/* toThrow */18](Jest.Expect[/* expectFn */1](Json.parseOrRaise, "{"));
                    }));
      }));

Jest.test("stringify", (function () {
        return Jest.Expect[/* toEqual */12]("null")(Jest.Expect[/* expect */0](Json.stringify(null)));
      }));

/*  Not a pure module */

File: __tests__/Json_test.ml
open Jest
open Expect
open Json

let _ =

describe "parse" (fun () ->
  test "success" (fun () ->
    expect @@
      parse "null"
      |> toEqual (Some Encode.null));

  test "error" (fun () ->
    expect @@
      parse "{"
      |> toEqual None);
);

describe "parseOrRaise" (fun () ->
  test "success" (fun () ->
    expect @@
      parseOrRaise "null"
      |> toEqual Encode.null);

  test "error" (fun () ->
    expectFn
      parseOrRaise "{"
      |> toThrowException (ParseError "Unexpected end of JSON input"));
);

test "stringify" (fun () ->
  expect @@
    stringify Encode.null
    |> toEqual "null");